### ISSUE TITLE
Regulating our MSRV

### ISSUE THREAD

**Issue by hdost** [open-telemetry/opentelemetry-rust#1554]
A recent Dependabot update changes [env_logger](https://github.com/rust-cli/env_logger)
The biggest risk from my perspective is the MSRV changed to 1.71

While we do have a policy that as long as we're X(To Be Filled) behind then we don't consider it breaking. However, I'd like us to consider that ferrocene is a certified compiler. Which can now be used in automotive use cases. Ferrocene is essentially rust but it's not the latest version. I will double check but from my recollection its latest is equivalent to Rust 1.65 (our current MSRV). 
There's also work on the rust project to deliver a spec for the language which should in theory develop a quicker cycle for getting certified in the future. 

What are your thoughts?

_Originally posted by @dependabot in https://github.com/open-telemetry/opentelemetry-rust/pull/1546_

**Comment by TommyCpp** [comment]
I think our general idea is to keep the MSRV as low as possible. Could you elaborate on what benifit of using a certified compiler? I'd image most of our users use vanilla Rust compiler

**Comment by hdost** [comment]
Yes i would agree that we don't want to bump the compiler if we do the have to, but that the ceiling for that should be lower than just 3 behind latest stable. 

The reason for using the certified compiler is for government regulation purposes, the most cited example is things like automotive. C and C++ follow compliance of a number of ISO standards. The goal for these is have predictable behavior from a spec. Currently the rust compiler itself doesn't have one (yet) but there's active work.

A blog post from late last year talks about some of it https://blog.rust-lang.org/inside-rust/2023/11/15/spec-vision.html
Ferrocene blog https://ferrous-systems.com/blog/officially-qualified-ferrocene/

**Comment by djc** [comment]
Note that env_logger is only used in examples, so keeping an older version around really doesn't matter. (Also, in many projects I have my MSRV CI jobs set to run `cargo check --lib` to avoid dev-dependencies like this.)

**Comment by hdost** [comment]
Ah so in this case we can also avoid this conflict for env_logger at least by not caring about the examples.
